src/subprocess.rs

```diff
@@ -1,32 +1,44 @@
-use std::os::unix::process::ExitStatusExt as _;
-
 /// An error representing a subprocess that errored
 ///
 /// This can be used to propogate a subprocesses exit status.
 /// When this error is encountered the cli will exit with the status code
 /// instead of printing an error,
 #[derive(Debug)]
-pub struct ExitStatusError {
-    status: Option<i32>,
+pub enum ExitStatusError {
+    ExitCode(i32),
+    Signal(i32),
+    Unknown,
 }
 
 impl ExitStatusError {
-    #[cfg(windows)]
+    #[cfg(unix)]
     pub(crate) fn new(status: std::process::ExitStatus) -> Self {
-        Self {
-            status: status.code(),
+        use std::os::unix::process::ExitStatusExt as _;
+
+        if let Some(code) = status.code() {
+            Self::ExitCode(code)
+        } else if let Some(signal) = status.signal() {
+            Self::Signal(signal)
+        } else {
+            Self::Unknown
         }
     }
 
-    #[cfg(unix)]
+    #[cfg(windows)]
     pub(crate) fn new(status: std::process::ExitStatus) -> Self {
-        Self {
-            status: status.code().or_else(|| status.signal()),
+        if let Some(code) = status.code() {
+            Self::ExitCode(code)
+        } else {
+            Self::Unknown
         }
     }
 
     pub fn code(&self) -> i32 {
-        self.status.unwrap_or(1)
+        match self {
+            Self::ExitCode(code) => *code,
+            Self::Signal(signal) => *signal,
+            Self::Unknown => 1,
+        }
```

Comment:
It would be good to at least try to avoid confusion between exit codes and signals, even though its only slightly within our control here.

https://tldp.org/LDP/abs/html/exitcodes.html
> 128+n Fatal error signal "n"
> 255 Exit status out of range
```suggestion
        // Exit code conventions: https://tldp.org/LDP/abs/html/exitcodes.html
        match self {
            Self::ExitCode(code) => *code,
            Self::Signal(signal) => 128 + *signal,
            Self::Unknown => 1,
        }.min(255)
```